### DISCUSSION TITLE
Auto remove roads build by AI

### DISCUSSION THREAD

**Discussion by axet** [OpenTTD/OpenTTD#10738] [Features]
Hello!

When AI compititors build rail roads, airports player can remove those by buying exclusive transport rights, and after few years AI player will be removed from town including airports, train stations and rail roads.

Which is not true for roads. Roads, bridges keept even after related business bus / track stations got removed. As result cities got polluted with a lot useless roads which block constructions and prevent you to expand.

Here is no way removing AI roads. Can roads be removed as soon as related bus / track stations removed by AI?

**Comment by PeterN** [comment]
I assume you mean after the company goes bankrupt, rather than when the AI removes stations itself.

This isn't done for roads because those roads could be used by anyone, but after a company goes bankrupt you should be able to remove them as they are no longer owned.

**Reply by axet** [reply]
I'm saying when I force AI remove station because station become not profitable. You can achieve that by buying exclusive transport rights for 3-5 years in a row.

**Reply by PeterN** [reply]
Well in that case, the AI removed the station itself, nothing was forced. And AI could be written to remove infrastructure it doesn't need, but that's a per AI thing.

Would you want your own town roads to be removed automatically just because you removed a station in the town?

**Reply by axet** [reply]
You probably never played for 2222+ year with AI. City's get very crowed with unused bridges and roads. Like A LOT!

**Comment by andythenorth** [comment]
Turn on the magic bulldozer (ctrl-alt-c to open cheat menu)

<img width="756" alt="image" src="https://user-images.githubusercontent.com/1780327/235302408-2ae6c5e6-d3fc-4a01-91da-22d514cb43ec.png">

**Reply by axet** [reply]
An option. But autoremove unused roads is better.

**Comment by andythenorth** [comment]
The simplest solution is to turn off the AI. Then this problem will not occur.  Thanks.

**Comment by TheMowgliMan** [comment]
This is true, but it would be unfair, as how would the code know if the
company is an AI or a player? They still own the roads, why remove them? A
better option would be for the AI to "sell" the roads to the district so
that they become public roads.
